### PR TITLE
feat(ActionBar): logic refactor

### PR DESCRIPTION
**Pull request by marcinsawicki**
Resolves: #1214 

## Description
Logic refactor:
- changed `IntersectionObserver` to `ResizeObserver` - from now logic will observe the container size changes, then will calculate how many option buttons will fit in the currently available space:
   - nothing is changing with buttons possible to show in the container
   - buttons for which there is not enough space will not be rendered (that was necessary only for `IntersectionObserver`) and will be available in the menu
- menu will be placed relative to the other elements (no more `absolute` positioning)

## Storybook

https://feature-action-bar-menu-icon-fix--613a8e945a5665003a05113b.chromatic.com

## Checklist

**Obligatory:**

- [x] Self review (use this as your final check for proposed changes before requesting the review)
- [x] Add reviewers (`livechat/design-system`)
- [x] Add correct label
- [x] Assign pull request with the correct issue

### CODE REVIEW

**Review by vladko-uxds — APPROVED** [review]
👍

File: packages/react-components/src/components/ActionBar/ActionBar.tsx [hunk 1]
```diff
@@ -24,128 +24,83 @@ export const ActionBar: React.FC<IActionBarProps> = ({
...
     styles[baseClass],
     className,
     vertical && styles[`${baseClass}--vertical`]
   );
-  const observerOptions = {
-    root: document.querySelector(`${id}`),
-    threshold: 1,
-  };
-  const shouldDisplayMenu = !isScrollType && menuItemsKeys.length !== 0;
+  const singleElementSize = 44;
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/ActionBar/ActionBar.tsx:38]: Shouldn't it be available as a prop? If not I'd suggest to move it outside the component's function.

File: packages/react-components/src/components/ActionBar/ActionBar.tsx [hunk 2]
```diff
@@ -24,128 +24,83 @@ export const ActionBar: React.FC<IActionBarProps> = ({
...
+    const hasIOSupport = !!window.ResizeObserver;
 
     if (!isScrollType && hasIOSupport) {
-      const target = document.querySelectorAll(
-        `button[data-actionbarid='${id}']`
-      );
+      const observer = new ResizeObserver((entries) => {
+        const { width, height } = entries[0].contentRect;
+        const containerSize = vertical ? height : width;
+        const exstraSpacing = menuOptions.length > 0 ? 60 : 0;
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/ActionBar/ActionBar.tsx:60]: `extraSpacing`

File: packages/react-components/src/components/ActionBar/ActionBar.tsx [hunk 3]
```diff
@@ -24,128 +24,83 @@ export const ActionBar: React.FC<IActionBarProps> = ({
...
-  }, [menuItemsKeys, isScrollType]);
-
-  const getMenuItems = (keys: string[]) => {
-    const filteredOptions = options.filter((row) =>
-      keys.find((k) => k === row.key && !row.hideInMenu)
-    );
+  }, [menuOptions, isScrollType, visibleItemsCount]);
 
-    return filteredOptions.map((o) => {
+  const getMenuItems = (menuOptions: IActionBarOption[]) =>
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/ActionBar/ActionBar.tsx:80]: useCallback?

**Review by VadymBezpalko — APPROVED** [review]
👍 
P.S.: Have you checked the component with React profiler to see how it renders?

File: packages/react-components/src/components/ActionBar/ActionBar.tsx (hunk 1 above)
Comment (marcinsawicki) [reply packages/react-components/src/components/ActionBar/ActionBar.tsx:38]: Currently not, but in the future I plan to improve that component and give a possibility to set the buttons size.

**Pull request merged**